### PR TITLE
Hotfix/summit

### PR DESCRIPTION
**Pull request by andre-merzky**
update summit configs, fix CUDA spec, dynamically switch between physical and logical GPU indexes.

This needs to be released quickly I'm afraid, as the GPU index fixes are needed for the COVID-19 effort.  Please give this a review today if you can, otherwise I'll release anyway, and we handle the fallout later.

### CODE REVIEW

File: src/radical/pilot/agent/scheduler/base.py [hunk 1]
```diff
@@ -833,73 +833,74 @@ def _try_allocation(self, unit):
...
+        # FIXME: This code should probably live elsewhere, not in this
+        #        performance critical scheduler base class
+        #
+        # FIXME: The specification for `CUDA_VISIBLE_DEVICES` is actually LM
+        #        dependent.  Assume the scheduler assigns the second GPU.
+        #        Manually, one would set `CVD=1`.  That also holds for launch
+        #        methods like `fork` which leave GPU indexes unaltered.  Other
+        #        launch methods like `jsrun` mask the system GPUs and only the
+        #        second GPU is visible, at all, to the task.  To CUDA the system
+        #        now seems to have only one GPU, and we need to be set `CVD=0`.
```
Comment (mturilli) [inline src/radical/pilot/agent/scheduler/base.py]: `we need to be set `CVD=0`.` -> we need to set it to `CVD=0`.

File: src/radical/pilot/agent/scheduler/base.py [hunk 2]
```diff
@@ -833,73 +833,74 @@ def _try_allocation(self, unit):
...
+        #
+        # FIXME: This code should probably live elsewhere, not in this
+        #        performance critical scheduler base class
+        #
+        # FIXME: The specification for `CUDA_VISIBLE_DEVICES` is actually LM
+        #        dependent.  Assume the scheduler assigns the second GPU.
+        #        Manually, one would set `CVD=1`.  That also holds for launch
+        #        methods like `fork` which leave GPU indexes unaltered.  Other
+        #        launch methods like `jsrun` mask the system GPUs and only the
+        #        second GPU is visible, at all, to the task.  To CUDA the system
```
Comment (mturilli) [inline src/radical/pilot/agent/scheduler/base.py]: I would remove `at all`

File: src/radical/pilot/agent/scheduler/base.py [hunk 3]
```diff
@@ -833,73 +833,74 @@ def _try_allocation(self, unit):
...
+        # Check if unit requires GPUs.  If so, set CUDA_VISIBLE_DEVICES to the
+        # list of assigned  GPU IDs.  We only handle uniform GPU setting for
+        # now, and will isse a warning on non-uniform ones.
+        #
+        # The default setting is ``
+        #
+        # FIXME: This code should probably live elsewhere, not in this
+        #        performance critical scheduler base class
+        #
+        # FIXME: The specification for `CUDA_VISIBLE_DEVICES` is actually LM
```
Comment (mturilli) [inline src/radical/pilot/agent/scheduler/base.py]: LM -> launch method (LM)

File: src/radical/pilot/agent/scheduler/base.py [hunk 4]
```diff
@@ -833,73 +833,74 @@ def _try_allocation(self, unit):
...
+                gpu_maps.append(node['gpu_map'])
+   
+        if not gpu_maps:
+            # no gpu maps, nothing to do
+            pass
+
+        elif len(gpu_maps) > 1:
+            # FIXME: this does not actually check for uniformity
+            self._log.warn('cannot set CUDA_VISIBLE_DEVICES for non-uniform'
+                           'GPU schedule (%s)' % gpu_maps)
```
Comment (mturilli) [inline src/radical/pilot/agent/scheduler/base.py]: 'cannot set CUDA_VISIBLE_DEVICES for non-uniform, the unit may fail!' 
'GPU schedule (%s)'

File: src/radical/pilot/agent/scheduler/base.py [hunk 5]
```diff
@@ -833,73 +833,74 @@ def _try_allocation(self, unit):
...
+
+        else:
+            gpu_map = gpu_maps[0]
+            if gpu_map:
+                # uniform, non-zero gpu map
+                if cvd_id_mode == 'physical':
+                    unit['description']['environment']['CUDA_VISIBLE_DEVICES']\
+                            = ','.join(str(gpu_set[0]) for gpu_set in gpu_map)
```
Comment (mturilli) [inline src/radical/pilot/agent/scheduler/base.py]: This is funny: `','.join(str(gpu_set[0]) for gpu_set in gpu_map)`. Does p3 have implicit list comprehension, i.e., we can omit `[]`?

**Comment by andre-merzky** [comment]
Thanks Matteo, I addressed the comments.

**Pull request merged**